app/assets/javascripts/admin/task/task_annotation_view.js

```diff
@@ -79,8 +85,14 @@ class TaskAnnotationView extends React.PureComponent<Props, State> {
 
   getDropdownMenu(annotation: APIAnnotationType) {
     // TODO use react fragments <> instead of spans
+    let doesAnnotationBelongToActiveUser = false;
+
+    if (annotation.user && this.props.activeUser) {
+      doesAnnotationBelongToActiveUser = annotation.user.id !== this.props.activeUser.id;
+    }
+
     const label =
-      annotation.state === "Finished" ? (
+      annotation.state === "Finished" || doesAnnotationBelongToActiveUser ? (
```

Comment:
I think some things are inverted here (probably these inversions cancel themselves out, which is why it works?):

- According to this line, the label reads "View" if the annotation belongs to the active user. However, it should read "View" if it belongs to **another** user.
- The condition in 91 should check for equality instead of inequality if the boolean is supposed to be true if the annotation belongs to the active user.

You probably want to rename the boolean to `doesAnnotationBelongToAnotherUser` (or `isNotOwnAnnotation`), so the semantics is right. Or keep the non negative name (or `isOwnAnnotation`) and invert its usages. Then, you need to initialize the boolean to true, though.

So, I count three inversions ^^ It's probably not working, when you are not logged in, since the default value of false will activate the "Trace" label.